src/sparsevector.jl

```diff
@@ -2138,11 +2138,14 @@ function _densifystarttolastnz!(x::SparseVector)
     x
 end
 
-#sorting
+#sorting TODO: integrate with `Base.Sort.IEEEFloatOptimization`'s partitioning by zero
+searchsortedfirst_discard_keywords(v::AbstractVector, x; lt=isless, by=identity,
+    rev::Union{Bool,Nothing}=nothing, order::Base.Order.Ordering=Forward, kws...) =
+        searchsortedfirst(v,x,Base.Order.ord(lt,by,rev,order))
 function sort!(x::AbstractCompressedVector; kws...)
```

Comment:
Why not apply the same trick to `sort!` directly?

Reply:
I want `sort!(sprand(10, .1), banana=:blue)` to continue to error.

Reply:
Makes sense. Judging from the code, however, I don't quite see where that call would throw.

Reply:
It should throw at `sort!(nz; kws...)`. I'll also add a test.